settings/defaultSettings.txt
baud:0
chlabel:0
clearonrec:1
csvinchid:0
csvios:0
csvprecisoin:5
csvvro:0
haxis:1
hdiv:-3
hrange:1000
lineending:1
linetimeout:18
math1en:0
math1first:1
math1op:0
math1sec:2
math2en:0
math2first:1
math2op:1
math2sec:2
math3en:0
math3first:1
math3op:2
math3sec:2
math4en:0
math4first:1
math4op:3
math4sec:2
mathios:0
mathvro:1
output:1
plottype:1
selused:1
vaxis:1
vdiv:-3
vpos:0
vrange:100
xyautosize:1
xyen:0
xyios:0
xyvro:0
xyxch:1
xyych:2
lang:en
csvdel:dc
ch:1:off:0
ch:1:sca:1
ch:1:inv:0
ch:1:sty:0
ch:1:name:
ch:1:col:255,0,0
ch:2:off:0
ch:2:sca:1
ch:2:inv:0
ch:2:sty:0
ch:2:name:
ch:2:col:0,0,255
ch:3:off:0
ch:3:sca:1
ch:3:inv:0
ch:3:sty:0
ch:3:name:
ch:3:col:255,170,0
ch:4:off:0
ch:4:sca:1
ch:4:inv:0
ch:4:sty:0
ch:4:name:
ch:4:col:85,170,0
ch:5:off:0
ch:5:sca:1
ch:5:inv:0
ch:5:sty:0
ch:5:name:
ch:5:col:170,0,255
ch:6:off:0
ch:6:sca:1
ch:6:inv:0
ch:6:sty:0
ch:6:name:
ch:6:col:85,0,0
ch:7:off:0
ch:7:sca:1
ch:7:inv:0
ch:7:sty:0
ch:7:name:
ch:7:col:170,170,0
ch:8:off:0
ch:8:sca:1
ch:8:inv:0
ch:8:sty:0
ch:8:name:
ch:8:col:255,85,127
ch:9:off:0
ch:9:sca:1
ch:9:inv:0
ch:9:sty:0
ch:9:name:
ch:9:col:133,167,37
ch:10:off:0
ch:10:sca:1
ch:10:inv:0
ch:10:sty:0
ch:10:name:
ch:10:col:24,1,182
ch:11:off:0
ch:11:sca:1
ch:11:inv:0
ch:11:sty:0
ch:11:name:
ch:11:col:59,103,230
ch:12:off:0
ch:12:sca:1
ch:12:inv:0
ch:12:sty:0
ch:12:name:
ch:12:col:134,242,52
ch:13:off:0
ch:13:sca:1
ch:13:inv:0
ch:13:sty:0
ch:13:name:
ch:13:col:207,54,54
ch:14:off:0
ch:14:sca:1
ch:14:inv:0
ch:14:sty:0
ch:14:name:
ch:14:col:114,126,43
ch:15:off:0
ch:15:sca:1
ch:15:inv:0
ch:15:sty:0
ch:15:name:
ch:15:col:136,180,244
ch:16:off:0
ch:16:sca:1
ch:16:inv:0
ch:16:sty:0
ch:16:name:
ch:16:col:104,177,97
ch:17:off:0
ch:17:sca:1
ch:17:inv:0
ch:17:sty:0
ch:17:name:
ch:17:col:81,65,18
ch:18:off:0
ch:18:sca:1
ch:18:inv:0
ch:18:sty:0
ch:18:name:
ch:18:col:217,145,231
ch:19:off:0
ch:19:sca:1
ch:19:inv:0
ch:19:sty:0
ch:19:name:
ch:19:col:98,200,193
ch:20:off:0
ch:20:sca:1
ch:20:inv:0
ch:20:sty:0
ch:20:name:
ch:20:col:105,48,117
ch:21:off:0
ch:21:sca:1
ch:21:inv:0
ch:21:sty:0
ch:21:name:
ch:21:col:139,246,96
ch:22:off:0
ch:22:sca:1
ch:22:inv:0
ch:22:sty:0
ch:22:name:
ch:22:col:238,211,56
ch:23:off:0
ch:23:sca:1
ch:23:inv:0
ch:23:sty:0
ch:23:name:
ch:23:col:0,85,127
ch:24:off:0
ch:24:sca:1
ch:24:inv:0
ch:24:sty:0
ch:24:name:
ch:24:col:191,135,193
ch:25:off:0
ch:25:sca:1
ch:25:inv:0
ch:25:sty:0
ch:25:name:
ch:25:col:161,88,235
ch:26:off:0
ch:26:sca:1
ch:26:inv:0
ch:26:sty:0
ch:26:name:
ch:26:col:61,14,226
ch:27:off:0
ch:27:sca:1
ch:27:inv:0
ch:27:sty:0
ch:27:name:
ch:27:col:69,83,74
ch:28:off:0
ch:28:sca:1
ch:28:inv:0
ch:28:sty:0
ch:28:name:
ch:28:col:255,170,127
ch:29:off:0
ch:29:sca:1
ch:29:inv:0
ch:29:sty:0
ch:29:name:
ch:29:col:118,35,27
ch:30:off:0
ch:30:sca:1
ch:30:inv:0
ch:30:sty:0
ch:30:name:
ch:30:col:81,99,20
ch:31:off:0
ch:31:sca:1
ch:31:inv:0
ch:31:sty:0
ch:31:name:
ch:31:col:85,170,127
ch:32:off:0
ch:32:sca:1
ch:32:inv:0
ch:32:sty:0
ch:32:name:
ch:32:col:106,76,194
ch:33:off:0
ch:33:sca:1
ch:33:inv:0
ch:33:sty:0
ch:33:name:
ch:33:col:83,154,130
ch:34:off:0
ch:34:sca:1
ch:34:inv:0
ch:34:sty:0
ch:34:name:
ch:34:col:173,181,89
ch:35:off:0
ch:35:sca:1
ch:35:inv:0
ch:35:sty:0
ch:35:name:
ch:35:col:137,151,202
ch:36:off:0
ch:36:sca:1
ch:36:inv:0
ch:36:sty:0
ch:36:name:
ch:36:col:170,0,127
ch:37:off:0
ch:37:sca:1
ch:37:inv:0
ch:37:sty:0
ch:37:name:
ch:37:col:134,245,70
ch:38:off:0
ch:38:sca:1
ch:38:inv:0
ch:38:sty:0
ch:38:name:
ch:38:col:92,178,70
ch:39:off:0
ch:39:sca:1
ch:39:inv:0
ch:39:sty:0
ch:39:name:
ch:39:col:201,63,43
ch:40:off:0
ch:40:sca:1
ch:40:inv:0
ch:40:sty:0
ch:40:name:
ch:40:col:170,118,212
ch:41:off:0
ch:41:sca:1
ch:41:inv:0
ch:41:sty:0
ch:41:name:
ch:41:col:140,66,74
ch:42:off:0
ch:42:sca:1
ch:42:inv:0
ch:42:sty:0
ch:42:name:
ch:42:col:202,51,26
ch:43:off:0
ch:43:sca:1
ch:43:inv:0
ch:43:sty:0
ch:43:name:
ch:43:col:194,67,215
ch:44:off:0
ch:44:sca:1
ch:44:inv:0
ch:44:sty:0
ch:44:name:
ch:44:col:82,141,27
ch:45:off:0
ch:45:sca:1
ch:45:inv:0
ch:45:sty:0
ch:45:name:
ch:45:col:217,52,237
ch:46:off:0
ch:46:sca:1
ch:46:inv:0
ch:46:sty:0
ch:46:name:
ch:46:col:59,84,224
ch:47:off:0
ch:47:sca:1
ch:47:inv:0
ch:47:sty:0
ch:47:name:
ch:47:col:82,10,203
ch:48:off:0
ch:48:sca:1
ch:48:inv:0
ch:48:sty:0
ch:48:name:
ch:48:col:128,214,174
ch:49:off:0
ch:49:sca:1
ch:49:inv:0
ch:49:sty:0
ch:49:name:
ch:49:col:76,116,56
ch:50:off:0
ch:50:sca:1
ch:50:inv:0
ch:50:sty:0
ch:50:name:
ch:50:col:151,218,140
ch:51:off:0
ch:51:sca:1
ch:51:inv:0
ch:51:sty:0
ch:51:name:
ch:51:col:194,68,130
ch:52:off:0
ch:52:sca:1
ch:52:inv:0
ch:52:sty:0
ch:52:name:
ch:52:col:212,45,173
ch:53:off:0
ch:53:sca:1
ch:53:inv:0
ch:53:sty:0
ch:53:name:
ch:53:col:120,243,185
ch:54:off:0
ch:54:sca:1
ch:54:inv:0
ch:54:sty:0
ch:54:name:
ch:54:col:72,129,25
ch:55:off:0
ch:55:sca:1
ch:55:inv:0
ch:55:sty:0
ch:55:name:
ch:55:col:64,116,225
ch:56:off:0
ch:56:sca:1
ch:56:inv:0
ch:56:sty:0
ch:56:name:
ch:56:col:156,248,126
ch:57:off:0
ch:57:sca:1
ch:57:inv:0
ch:57:sty:0
ch:57:name:
ch:57:col:175,61,153
ch:58:off:0
ch:58:sca:1
ch:58:inv:0
ch:58:sty:0
ch:58:name:
ch:58:col:215,221,87
ch:59:off:0
ch:59:sca:1
ch:59:inv:0
ch:59:sty:0
ch:59:name:
ch:59:col:103,76,208
ch:60:off:0
ch:60:sca:1
ch:60:inv:0
ch:60:sty:0
ch:60:name:
ch:60:col:170,170,127
ch:61:off:0
ch:61:sca:1
ch:61:inv:0
ch:61:sty:0
ch:61:name:
ch:61:col:85,0,0
ch:62:off:0
ch:62:sca:1
ch:62:inv:0
ch:62:sty:0
ch:62:name:
ch:62:col:85,85,127
ch:63:off:0
ch:63:sca:1
ch:63:inv:0
ch:63:sty:0
ch:63:name:
ch:63:col:242,190,90
ch:64:off:0
ch:64:sca:1
ch:64:inv:0
ch:64:sty:0
ch:64:name:
ch:64:col:87,146,229
ch:65:off:0
ch:65:sca:1
ch:65:inv:0
ch:65:sty:0
ch:65:name:Math 1
ch:65:col:0,0,0
ch:66:off:0
ch:66:sca:1
ch:66:inv:0
ch:66:sty:0
ch:66:name:Math 2
ch:66:col:0,0,0
ch:67:off:0
ch:67:sca:1
ch:67:inv:0
ch:67:sty:0
ch:67:name:Math 3
ch:67:col:0,0,0
ch:68:off:0
ch:68:sca:1
ch:68:inv:0
ch:68:sty:0
ch:68:name:Math 4
ch:68:col:0,0,0


settings.h
#ifndef SETTINGS_H
#define SETTINGS_H

#include <QColor>
#include <QObject>
#include <QVector>

enum PlotStatus { run, pause, single };

enum PlotRange { freeMove, fixed, rolling };

enum GraphStyle { line, point, linePoint, hidden };

enum DataMode { unknown, string, bin, terminal, info, warning };

enum OutputLevel { none, low, high };

enum MathOperations { add, subtract, multiply, divide, xy };

enum DataLineType { command, data, timeouted };

#define CHANNEL_COUNT 64

#define CMD_BEGIN "<cmd>"
#define TIMEOUT_SYMBOL "<timeout>"
#define CMD_END "<end>"

#define PORT_NUCLEO_DESCRIPTION_IDENTIFIER "ST"

#define POINT_STYLE QCPScatterStyle::ssDisc

const QString lineEndings[4] = {"", "\n", "\r", "\r\n"};

//                             0      1      2      3      4      5      6       7
const float refreshRates[8] = {10.0f, 20.0f, 30.0f, 40.0f, 50.0f, 60.0f, 100.0f, 144.0f};

//                                     0       1       2       3      4      5      6     7     8     9    10   11  12 13 14  15  16  17  18   19   20   21    22    23    24     25     26     27      28      29      30
const double logaritmicSettings[31] = {0.0001, 0.0002, 0.0005, 0.001, 0.002, 0.005, 0.01, 0.02, 0.05, 0.1, 0.2, 0.5, 1, 2, 5, 10, 20, 50, 100, 200, 500, 1000, 2000, 5000, 10000, 20000, 50000, 100000, 200000, 500000, 1000000};

const QColor defaultColors[8] = {Qt::darkRed, Qt::darkBlue, Qt::darkGreen, Qt::darkYellow, Qt::darkMagenta, Qt::darkCyan, Qt::darkGray, Qt::black};

struct BinDataSettings_t {
  int bits = 8;
  double valueMin = 0;
  double valueMax = 255;
  double timeStep = 1;
  int numCh = 1;
  int firstCh = 1;
  bool continuous = false;
};

struct ChannelSettings_t {
  QColor color = QColor(Qt::black);
  int style = GraphStyle::line;
};

struct PlotSettings_t {
  double rollingRange = 100;
  int zoom = 1000;
  double horizontalPos = 500;
  double verticalRange = 10;
  int verticalCenter = 0;
};

Q_DECLARE_METATYPE(BinDataSettings_t)
Q_DECLARE_METATYPE(ChannelSettings_t)
Q_DECLARE_METATYPE(PlotSettings_t)

#define MIN(a, b) (((a) < (b)) ? (a) : (b))
#define MAX(a, b) (((a) > (b)) ? (a) : (b))

#define CMD_END_LENGTH (sizeof(CMD_END) - 1)
#define TIMEOUT_SYMBOL_LENGTH (sizeof(TIMEOUT_SYMBOL) - 1)
#define CMD_BEGIN_LENGTH (sizeof(CMD_BEGIN) - 1)

#endif // SETTINGS_H
